Code/Framework/AzCore/AzCore/std/string/conversions.h

```diff
@@ -266,6 +306,41 @@ namespace AZStd
         return to_string(dest, src.c_str(), src.length());
     }
 
+    template<size_t MaxElementCount>
+    void to_string(AZStd::basic_fixed_string<string::value_type, MaxElementCount, string::traits_type>& dest, const wchar_t* str, size_t srcLen = 0)
+    {
+        dest.clear();
+
+        if (srcLen == 0)
+        {
+            srcLen = wcslen(str);
+        }
+
+        if (srcLen > 0)
+        {
+            Internal::WCharTPlatformConverter<>::to_string(dest, str, str + srcLen);
+        }
+    }
+
+    template<size_t MaxElementCount1, size_t MaxElementCount2>
+    void to_string(AZStd::basic_fixed_string<string::value_type, MaxElementCount1, string::traits_type>& dest, const AZStd::basic_fixed_string<wstring::value_type, MaxElementCount2, wstring::traits_type>& src)
+    {
+        return to_string(dest, src.c_str(), src.length());
+    }
+
+    inline void to_string(char* dest, size_t destSize, const wchar_t* str, size_t srcLen = 0)
+    {
+        if (srcLen == 0)
+        {
+            srcLen = wcslen(str);
+        }
+
+        if (srcLen > 0)
+        {
+            Internal::WCharTPlatformConverter<>::to_string(dest, destSize, str, str + srcLen + 1); // copy null terminator
```

Comment:
We should not copy pass the srcLen.
There could be an invalid assumption that the source path is null-terminated.
The null-terminating character should set after the wchar is converted to a char*
```suggestion
            Internal::WCharTPlatformConverter<>::to_string(dest, destSize, str, str + srcLen);
            dest[srcLen] = '\0';
```

Reply:
its more tricky than that, `srcLen` is not the same length that will have at the destination
In this case, when converting to_string, the destination can have some unicode characters that are taking more than one position to encode. However, I could return the last position on dest from the Internal::WCharTPlatformConverter method to get a pointer to there

Reply:
its more tricky than that, srcLen is not the same length that will have at the destination
In this case, when converting to_string, the destination can have some unicode characters that are taking more than one position to encode. However, I could return the last position on dest from the Internal::WCharTPlatformConverter method to get a pointer to there

Reply:
I think returning the number of "characters" written would be useful.

Reply:
the signature of the conversion methods returns a pointer to the the end of the source so I used that

Reply:
Good enough for me.